design/list_command_design.md

```diff
@@ -0,0 +1,272 @@
+# Dockerless CLI - `list` action
+
+This document is the design document for the *list* CLI command for the Dockerless project. The main functionality for the list CLI command has been implemented. Now all that is left is to implement the following flags:
+
+1. `-i, --inspect`
+2. `-k, --kind`
+3. `-l, --limit`
+5. `-r, --role`
+6. `-s, --search`
+6. `-o, --offset` (low in priority)
+
+`offset` is at a lower priority because the current Ruby CLI does not implement this flag.
+
+In additional to implementing the above flags, we will need to also address the following:
+
+1. Ensure UTs and integration tests exist and are GA quality
+3. Write proper documentation in `README.md` of the repository
+
+For the full UX design for the list command see [here](https://ljfz3b.axshare.com/#id=yokln4&p=conjur_main_help&g=1).
+
+### UX mapping
+
+We use a third-party module called [argparse](https://docs.python.org/3/library/argparse.html) which gives us limited control over the types of errors we can return. Therefore, at this time we will not surgicially change the errors and leave them as is. The following is a mapping of the current flow for all possible permutations of the *list* CLI command and their expected outputs/behavior. 
+
+*When no values were found*
+
+```bash
+$ conjur-cli list
+
+## Current
+[]
+
+## Should be
+No results found
+```
+
+*List with more inputs than excepted*
+
+```bash
+$ conjur-cli list unknown
+
+## Current
+usage: conjur-cli [-h] [-v] [-l URL] [-a ACCOUNT] [-c CA_BUNDLE] [--insecure] [-u LOGIN_ID] [-k API_KEY] [-p PASSWORD] [-d]
+         [--verbose]
+         {list,variable,policy} ...
+conjur-cli: error: unrecognized arguments: boo
+
+## Should return an error
+```
+
+*Mispell list command*
+
+```bash
+$ conjur-cli listboo
+
+## Current
+usage: conjur-cli [-h] [-v] [-l URL] [-a ACCOUNT] [-c CA_BUNDLE] [--insecure] [-u LOGIN_ID] [-k API_KEY] [-p PASSWORD] [-d]
+                  [--verbose]
+                  {list,variable,policy} ...
+conjur-cli: error: argument resource: invalid choice: 'listl' (choose from 'list', 'variable', 'policy')
+
+## Should be:
+conjur-cli: error: argument resource: invalid choice: 'listn' (choose from 'list', 'variable', 'policy')
```

Comment:
you meant invalid choice: 'listboo', right?

Reply:
💯